### DISCUSSION TITLE
Error when installing the agent with quiet mode

### DISCUSSION THREAD

**Discussion by Alefunshow** [glpi-project/glpi-agent#272] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

Yes, I know

### Describe the bug

Hi, I'm trying the vbs script for the installation of the agent via gpo. I'm unable to install it until i remove the /quiet mode. Is there any other argument that I have to use with the /quiet mode because I can't install 350 agent with the UI mode

### To reproduce

Installing glpi agent with quiet mode on

### Expected behavior

I'm expecting to install glpi agent with quiet mode

### Operating system

Windows

### GLPI Agent version

1.4

### GLPI version

10.0.3

### GLPIInventory plugin or FusionInventory for GLPI plugin version

10.0.1+1.0

**Comment by g-bougard** [comment]
Please provide the Options value you're using. You probably missed something.

**Comment by Alefunshow** [comment]
@g-bougard  /quiet RUNNOW=1 SERVER='http://*****.it/plugins/fusioninventory/' HTTPD_TRUST=127.0.0.1/32

**Comment by g-bougard** [comment]
Do you run the script in a PowerShell context ?
You should check your system log event viewer for error related to the msi installation.
You can also try to add `/L*V 'C:\agent-install.log'` to generate a MSI exec log and see what error is reported in that log.

**Comment by Alefunshow** [comment]
It doesn't write the log file

**Comment by Alefunshow** [comment]
@g-bougard This is the code that I edited. I tried to run this either on cmd and powershell with and without administrator privileges. No chance to get the installation done.
```
'
'
' USER SETTINGS
'
'

' SetupVersion
'    Setup version with the pattern <major>.<minor>.<release>[-<package>]
'
SetupVersion = "1.4"

' When using a nightly built version, uncomment the following SetupVersion definition line
' replacing gitABCDEFGH with the most recent git revision found on the nightly builds site
' In that case, SetupNightlyLocation will be selected as location in place of SetupLocation
'SetupVersion = "1.0-gitABCDEFGH"

' SetupLocation
'    Depending on your needs or your environment, you can use either a HTTP or
'    CIFS/SMB.
'
'    If you use HTTP, please, set to SetupLocation a URL:
'
'       SetupLocation = "http://host[:port]/[absolut_path]" or
'       SetupLocation = "https://host[:port]/[absolut_path]"
'
'    If you use CIFS, please, set to SetupLocation a UNC path name:
'
'       SetupLocation = "\\host\share\[path]"
'
'       You also must be sure that you have removed the "Open File Security Warning"
'       from programs accessed from that UNC.
'
' Location for Release Candidates
'SetupLocation = "https://github.com/glpi-project/glpi-agent/releases/download/" & SetupVersion
SetupLocation = "\\mainsrv\fusioninventory"

' Location for Nightly Builds
'SetupNightlyLocation = "https://nightly.glpi-project.org/glpi-agent"


' SetupArchitecture
'    The setup architecture can be 'x86', 'x64' or 'Auto'
'
'    If you set SetupArchitecture = "Auto" be sure that both installers are in
'    the same SetupLocation.
'
SetupArchitecture = "Auto"

' SetupOptions
'    Consult the online installer documentation to know its list of options.
'    See: https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html#command-line-parameters
'
'    You should use simple quotes (') to set between quotation marks those values
'    that require it; double quotes (") doesn't work with UNCs.
'
' SetupOptions = " /i /quiet RUNNOW=1 SERVER='http://glpi.yourcompany.com/'"
SetupOptions = "/quiet RUNNOW=1 SERVER='http://******.it/plugins/fusioninventory/' HTTPD_TRUST='127.0.0.1/32,*****.it'"

' Setup
'    The installer file name. You should not have to modify this variable ever.
'
Setup = "GLPI-Agent-" & SetupVersion & "-" & SetupArchitecture & ".msi"

' Reconfigure
'    Just reconfigure the current installation if installed agent has the same version
'
Reconfigure = "No"

' Repair
'    Repair the installation when Setup is still installed.
'
Repair = "No"

' Verbose
'    Enable or disable the information messages.
'
'    It's advisable to use Verbose = "Yes" with 'cscript //nologo ...'.
'
Verbose = "No"

' RunUninstallFusionInventoryAgent
'    Set to "Yes" to first uninstall FusionInventory Agent
'    Also and unless SERVER or LOCAL are defined in SetupOptions, this script
'    will try to get them from FusionInventory-Agent configuration found in registry
'
RunUninstallFusionInventoryAgent = "Yes"

' UninstallOcsAgent
'    Enable or disable the uninstallation of OCS Agent
'
UninstallOcsAgent = "Yes"
```

**Comment by g-bougard** [comment]
Your SetupLocation seems to be a shared disk. Are you sure it is available and readable when the script is started ?
Have you checked the system eventviewer as I suggested ?

**Comment by Alefunshow** [comment]
@g-bougard As I said yesterday the location is available in fact if I remove the quiet mode it works. With quiet mode (needed for the amount of agent in my network; 350 agent) it doesn't work. The event of windows shows nothing.

**Comment by g-bougard** [comment]
Actually, I don't see any bug. You should probably request help on a forum more dedicated to windows admin and MSI installation.
I'll keep this issue opened if somebody from the community can help. But I'll close it in a week if nobody comments.

You may make a try without the mount point to check if it's working when downloading from internet.

**Comment by keytrap-x86** [comment]
Hello @Alefunshow 
I have deployed the glpi agent on +200 computers in my domain network.
Here's what I have in my NETLOGON folder:
![image](https://user-images.githubusercontent.com/17864005/203528929-f13684f7-6707-4573-ac61-523c61fff91d.png)

**install.cmd** :

```
"c:\Program Files\FusionInventory-Agent\Uninstall.exe" /S
"c:\Program Files (x86)\FusionInventory-Agent\Uninstall.exe" /S
cscript //nologo %~dp0glpi-agent-deployment.vbs
```
(don't remove the `%~dp0`, it means "current directory")

The first two lines are for uninstalling fusion inventory before installing the glpi agent.
In the **[glpi-agent-deployment.vbs](https://raw.githubusercontent.com/glpi-project/glpi-agent/develop/contrib/windows/glpi-agent-deployment.vbs)** I only changed the line 118:

![image](https://user-images.githubusercontent.com/17864005/203530184-fe1ef65f-4e27-4597-8b4e-f5ea7eb0cee2.png)

The **install.cmd** script is launched with a GPO at computer startup :

![image](https://user-images.githubusercontent.com/17864005/203530444-6ebd9f08-d0e0-473a-8254-f558dbae3069.png)

Hopefully this will help you out

**Comment by keytrap-x86** [comment]
By the way if you are installing the glpi-agent you should not use `http://******.it/plugins/fusioninventory/` in your server url but `http://******.it/marketplace/glpiinventory` and you have to install the glpi agent plugin with the market place

![image](https://user-images.githubusercontent.com/17864005/203532071-45b69f23-bbd3-4e2b-8ce5-9457494ed384.png)

**Reply by kintaro1981** [reply]
I don't think this is mandatory, you can choose how to install it (manually in plugins directory) or from the marketplace. right?

**Reply by g-bougard** [reply]
I confirm: the plugin can be installed manually under `/plugins/glpiinventory`.
